app/src/components/CustomPeriodDialog.tsx

```diff
@@ -25,16 +25,20 @@ export function CustomPeriodDialog(props: CustomPeriodDialogProps) {
   const [endTime, setEndTime] = useState<TimeValue>(new Time(12, 0));
 
   const [startDate, setStartDate] = useState<DateValue>(
-    toCalendarDate(new Date(Date.now() - PeriodProps[Period.WEEK].milliseconds))
+      toCalendarDate(new Date(
+          Math.max(Date.now() -
+              PeriodProps[Period.WEEK].milliseconds,
+              new Date('2013-01-01T00:00:00').getTime())))
   );
 
   const [endDate, setEndDate] = useState<DateValue>(toCalendarDate(new Date()));
 
   function handleSelection() {
-    const startDateTime = toDate(startDate, startTime);
+    const startDateTime = Math.max(toDate(startDate, startTime).getTime(),
```

Comment:
Could you move this 2013 date variable to a constant above the component function? `MIN_DATE` or something, and maybe a comment explaining why 2013 (it might not be obvious to everyone reading)